content/blog/2022-03-29-unified-gc/index.md

```diff
@@ -0,0 +1,115 @@
+
+## Fix-Point Formulation
+
+Garbage collection is first defined using a fixed point formulation. Later in the paper, it is shown how tracing, reference counting, and hybrids of tracing and reference counting satisfy this fixed point formulation. We start out with a characterization of memory:
+* $V$ is the set of all objects. This includes objects that are still in use, objects that are no longer in use but not yet freed, and objects that have been freed. 
+* $E$ is the multiset of edges in the graph. In other words, the references between one object and another. It is a multi-set because an object can have multiple pointers to another node. Consider the case where object $a$ has fields $f1$ and $f2$ which both point to the same object $b$.
+* $R$ is the multiset of objects that are roots in the graph.
+* $p(v)$ where $v \in V$ is the reference count of vertex $v$.
+
+The object graph is the triple $G = <V, E, R>$. From here, garbage collection is given as a fixed point computation.
+
+<p align="center">
+<img src="FP.png" alt="alt_text" title="image_tooltip" style="zoom:25%;" />
+</p>
+
+Once reference counts have been assigned, vertices with counts of 0 are reclaimed. It is important to note that this fix point equation is not an algorithm. Instead, tracing, reference counting, and hybrids of the two are algorithms that satisfy this fix point equation.
+
+## Algorithmic Duals
+
+One of the key insights of this paper was that tracing and reference counting garbage collection methods are "algorithmic duals" of one another. The paper gives the intuition that tracing opperate on live objects or "matter" while reference counting operates on dead objects or "anti-matter".
+
+## Tracing vs Reference Counting
+
+The tracing garbage collection and reference counting algorithms are shown below. Tracing computes the least fix point and reference counting computes the greatest fix point. The set difference between the two fix point solutions is comprised of cyclical garbage. Cyclical garbage will be discussed in more detail below.
+
+<p align="center">
+<img src="TAndRC.png" alt="alt_text" title="image_tooltip" style="zoom:25%;" />
+</p>
+
+Tracing initially sets the reference counts of all objects to 0 and initializes the worklist to be the root set $R$. The $scan-by-tracing()$ function scans through the worklist and increments the reference counts of the objects it encounters, and adds all of the objects referenced by it to the worklist to be recursively processed as well. When this function terminates, the reference counts for live objects will have non-zero counts and all other objects will have a count of zero. From here the $sweep-for-tracing()$ function can free all objects of count zero so their memory can be reused.
```

Comment:
Maybe you meant to use `code` instead of $math$ for scan-by-tracing() and similar in the below paragraphs? It looks a little weird because the -s turn into minus signs in math mode. (Try previewing your post with math rendered.)